include/sys/arc_impl.h

```diff
@@ -995,7 +995,7 @@ extern void arc_wait_for_eviction(uint64_t, boolean_t);
 
 extern void arc_lowmem_init(void);
 extern void arc_lowmem_fini(void);
-extern void arc_prune_async(int64_t);
+extern void arc_prune_async(uint64_t);
```

Comment:
It seems arc_evict_state() may have a race when this argument may become negative,

Reply:
That would imply that there is currently an issue on Linux since this is assigned to an unsigned variable in the Linux version at `ap->p_adjust = adjust;`. I will study this more after I look at your other remarks.

Reply:
Are you referring to `aggsum_compare()` saying that the value is bigger than `arc_dnode_size_limit`, but then the value changing under us when we call aggsum_upper_bound() to something smaller than `arc_dnode_size_limit`?

`aggsum_upper_bound()` returns a `uint64_t` and `arc_dnode_size_limit` is a `uint64_t`, so if this happens, we should have integer underflow in the existing code. This patch does not make things any worse.

Reply:
Also, while I imagine we should address this like we did in ccec88f11a44746f78f88aac90f5172a52e04506, I am not seeing a way of addressing this that I particularly like. Suggestions would be appreciated. :)